src/plugin/plugin.ts

```diff
@@ -817,4 +818,52 @@ export default class Plugin {
     this.unloadedRcvdPromise = new Deferred();
     this.sendMsg(MessageType.PLUGIN_UNLOAD_REQUEST, {});
   }
+
+  unloadComponents(): Promise<void> {
+    const adapters = Array.from(this.adapters.values());
+    const notifiers = Array.from(this.notifiers.values());
+    const apiHandlers = Array.from(this.apiHandlers.values());
+
+    const unloadPromises = [];
+
+    for (const [, adapter] of this.adapters.entries()) {
```

Comment:
`for (const adapter of this.adapters.values())` would be cleaner. Ditto for the notifier block below.

Reply:
Done